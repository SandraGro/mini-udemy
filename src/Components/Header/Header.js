import React, { Component } from "react";
import "./Header.scss";
import {
  Form,
  Nav,
  Navbar,
  FormControl,
  NavDropdown,
  Button,
} from "react-bootstrap";

class Header extends Component {
  render() {
    // const categories2 = [];
    // for (let i in this.props.featuredCourses) {
    //   const categoriesDuplicated = categories.filter(
    //     (category) => category === this.props.featuredCourses[i].category
    //   );
    //   if (!(categoriesDuplicated.length > 0)) {
    //     categories2.push(this.props.featuredCourses[i].category);
    //   }
    // }
    const categories = {};
    for (let course of this.props.featuredCourses) {
      categories[course.category] = true;
    }

    return (
      <Navbar bg="light" expand="lg" className="main-navbar navbar">
        <Navbar.Brand href="/">Udemyx</Navbar.Brand>
        <Navbar.Toggle aria-controls="basic-navbar-nav" />
        <Navbar.Collapse id="basic-navbar-nav">
          <Nav className="justify-content-end mr-auto">
            <NavDropdown title="Categories" id="basic-nav-dropdown">
              {Object.keys(categories).map((category, index) => (
                <NavDropdown.Item
                  key={`dropdown-option-${index}`}
                  href="#action/3.1"
                >
                  {category}
                </NavDropdown.Item>
              ))}
              <NavDropdown.Item href="#action/3.4">
                All courses
              </NavDropdown.Item>
            </NavDropdown>
            <Form inline>
              <FormControl
                type="text"
                placeholder="Search"
                className="input-search mr-sm-2"
              />
            </Form>
            <Nav.Link href="#home">Udemyx for Business </Nav.Link>
            <Nav.Link href="#home">Teach on Udemyx </Nav.Link>
            <Nav.Link href="#link">My courses</Nav.Link>
            <Button variant="outline-success">Log in</Button>
            <Button variant="success">Sign up</Button>
          </Nav>
        </Navbar.Collapse>
      </Navbar>
    );
  }
}
export default Header;
