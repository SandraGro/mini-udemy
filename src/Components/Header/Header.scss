.main-navbar.navbar {
    height: 100px;

    .input-search {
        width: 795px;
        height: 45px;
        border-radius: 25px;
    }
}