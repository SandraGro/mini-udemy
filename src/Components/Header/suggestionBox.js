import React from "react";
import "./suggestionBox.scss";
import { Link } from "react-router-dom";

function SuggestionBox(props) {
  const { suggestions } = props;
  return (
    <ul
      className={`${
        suggestions.length ? "visible " : ""
      }suggestion-box suggestion-box-list`}
    >
      {suggestions.map((suggestion, index) => (
        <Link key={`suggestionItem-${index}`}  to={`/course/${suggestion.slug}`}>
          <li key={`suggestion-${index}`}>
            <span>{suggestion.category} | </span>
            <span>{suggestion.title}. </span>
            <span>{suggestion.author}</span>
            <hr></hr>
          </li>
        </Link>
      ))}
    </ul>
  );
}

export default SuggestionBox;
