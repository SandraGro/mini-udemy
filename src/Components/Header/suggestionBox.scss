.suggestion-box{
    animation: udlite-search-form-autocomplete-expand 150ms cubic-bezier(0,0,.38,.9);
    background: #fff;
    box-sizing: border-box;
    border: 1px solid #dcdacb;
    border-radius: 4px;
    padding: 1.6rem;
    position: absolute;
    transform-origin: top;
    z-index: 1000;
    visibility: hidden;
    width: 300px;
    font-weight: 400;
    line-height: 1.4;
    font-size: 14px;
    &.visible {
        visibility: visible;
    }
}

.suggestion-box-list{
    list-style: none;
    margin: 0;

}
