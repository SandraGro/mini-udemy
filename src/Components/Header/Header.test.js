import React from "react";
import { shallow, mount } from "enzyme";
import { BrowserRouter as Router } from "react-router-dom";
import toJson from "enzyme-to-json";

import Header from "./Header";

describe("Header component tests", () => {
  const user = {
    cart: [],
    myCourses: [],
    id: "user01",
    username: "Sandra",
  };
  const featuredCourses = [];
  const searchCourse = function searchCourse(searchTerm) {
    return [];
  };
  const history = { push: function push() {} };
  it("should render correctly", () => {
    const header = shallow(
      <Router>
        <Header
          history={history}
          user={user}
          featuredCourses={featuredCourses}
          searchCourse={searchCourse}
        />
      </Router>
    );
    expect(toJson(header)).toMatchSnapshot();
  });
  it("should render the correct count on the cart icon", () => {
      //arrange
    const testUser = {
      ...user,
      cart: [
        { title: "title1", author: "author1" },
        { title: "title2", author: "author2" },
        { title: "title3", author: "author3" },
      ],
    };
    const header = mount(
      <Router>
        <Header
          history={history}
          user={testUser}
          featuredCourses={featuredCourses}
          searchCourse={searchCourse}
        />
      </Router>
    );
    // Find badge element and get HTML
    // Make sure that selector returns an element so HTML function works

    //act
    const badge = header.find(".badge.badge-danger");
    //assert
    expect(badge.html()).toBe(`<span class="badge badge-danger">3</span>`);
  });
});
