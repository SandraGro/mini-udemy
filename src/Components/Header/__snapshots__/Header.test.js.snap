// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component tests should render correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Header
    featuredCourses={Array []}
    history={
      Object {
        "push": [Function],
      }
    }
    searchCourse={[Function]}
    user={
      Object {
        "cart": Array [],
        "id": "user01",
        "myCourses": Array [],
        "username": "Sandra",
      }
    }
  />
</Router>
`;
